src/classes/ACCT_IndividualAccounts_TDTM.cls

```diff
@@ -265,13 +273,25 @@ public class ACCT_IndividualAccounts_TDTM extends TDTM_Runnable {
                 primaryContactByAccountId.put(contactRecord.AccountId, contactRecord.Id);
             }
 
-            if (isHouseholdMember(contactRecord)) {
+            if ((!isNewContactInsertOperation() ||
+                    isNewContactInsertOperation() && isBeingAddedToAHousehold(contactRecord)) &&
+                    isHouseholdMember(contactRecord)) {
                 householdIdsToRename.add(contactRecord.AccountId);
             }
         }
 
         updatePrimaryContactOnAccount(primaryContactByAccountId);
-        householdNamingService.renameHouseholdAccounts(householdIdsToRename);
+        if (!householdIdsToRename.isEmpty()) {
+            householdNamingService.renameHouseholdAccounts(householdIdsToRename);
+        }
+    }
+
+    private Boolean isBeingAddedToAHousehold(Contact contactRecord) {
+        return contactRecord.AccountId != null;
+    }
+
+    private Boolean isNewContactInsertOperation() {
```

Comment:
I think I see why there is a method for `isNewContactInsertOperation` as well as a var of `isConInsert` (that I recommended renaming above), however it's not fully clear without spending time looking at the code. I'm not really sure that this method needs a var for `isConInsert`. As an instance method, it could just check for `(this.triggerAction == TDTM_Runnable.Action.BeforeInsert || this.triggerAction == TDTM_Runnable.Action.AfterInsert)` to make the same determination.

Reply:
The member variable "isNewContactInsert" prevents recursion when this class has kicked off the household account insert operation.The reference to the same field on the other class prevents recursion when that one kicked off the insert of Household objects.  I left a comment to clarify.